webapi/src/test/resources/test-data/searchR2RV2/bookWithIncomingPagesOnlyLink.jsonld

```diff
@@ -17,11 +17,13 @@
       "@type" : "knora-api:LinkValue",
       "knora-api:linkValueHasSource" : {
         "@id" : "http://rdfh.ch/50e7460a7203",
-        "@type" : "incunabula:page",
-        "incunabula:partOfValue" : {
+        "@type" : "p0803-incunabula:page",
+        "p0803-incunabula:partOfValue" : {
           "@id" : "http://rdfh.ch/50e7460a7203/values/8bdc04c8-b765-44c8-adb3-5ab536dcd051",
           "@type" : "knora-api:LinkValue",
-          "knora-api:linkValueHasTargetIri" : "http://rdfh.ch/8be1b7cf7103"
+          "knora-api:linkValueHasTargetIri" : {
+            "@id" : "http://rdfh.ch/8be1b7cf7103"
```

Comment:
@tobiasschweizer When I changed the way IRIs are represented in JSON-LD, I missed `linkValueHasSourceIri` and `linkValueHasTargetIri`. It's fixed here. Does this require a change in SALSAH?

Reply:
Yes, I think so. 

`src/app/model/webapi/knora/v2/convert-jsonld.ts`

```typescript
case AppConfig.LinkValue:

                let linkValue: ReadLinkValue;

                // check if the referred resource is given as an object or just as an IRI
                if (propValue[AppConfig.linkValueHasTarget] !== undefined) {
                    // linkValueHasTarget contains the object

                    let referredResource: ReadResource = constructReadResource(propValue[AppConfig.linkValueHasTarget]);

                    linkValue = new ReadLinkValue(propValue['@id'], propIri, referredResource.id, referredResource);
                } else if (propValue[AppConfig.linkValueHasTargetIri] !== undefined) {
                    // linkValueHasTargetIri contains the resource's Iri

                    let referredResourceIri = propValue[AppConfig.linkValueHasTargetIri];

                    linkValue = new ReadLinkValue(propValue['@id'], propIri, referredResourceIri);
                } else if (propValue[AppConfig.linkValueHasSource] !== undefined) {
                    // linkValueHasSource contains the object

                    let incomingResource: ReadResource = constructReadResource(propValue[AppConfig.linkValueHasSource]);

                    linkValue = new ReadLinkValue(propValue['@id'], propIri, incomingResource.id, incomingResource);
                } else if (propValue[AppConfig.linkValueHasSourceIri] !== undefined) {
                    // linkValueHasSourceIri contains the resource's Iri

                    let incomingResourceIri = propValue[AppConfig.linkValueHasSourceIri];

                    linkValue = new ReadLinkValue(propValue['@id'], propIri, incomingResourceIri);
                }

                valueSpecificProp = linkValue;
                break;
```

This is an object too now, right? So  we need to add `['@id']`. A test should be added to Salsah (on branch `unit tests`).

I can try to do it this afternoon.

Reply:
Do you have a test response in JDON-LD complex I could use as a test case?

I think the Iris are only returned in complex schema if there are circular references.

Reply:
Thanks, sorry about that!

The test responses are:

* `webapi/src/test/resources/test-data/searchR2RV2/bookWithIncomingPagesOnlyLink.jsonld`
* `webapi/src/test/resources/test-data/searchR2RV2/bookWithIncomingPagesWithAllRequestedProps.jsonld`

Reply:
done in https://github.com/dhlab-basel/Salsah/pull/247/commits/074ae6a7a5366bffb7ee44a517ad5fe72b2dffef

Reply:
Thank you!